src/Utilities/MuxLogger.cs

```diff
@@ -321,7 +321,11 @@ private void ProjectFinished(object sender, ProjectFinishedEventArgs e)
         /// <summary>
         /// This class holds everything the logger needs to know about a particular submission, including the event source.
         /// </summary>
-        private class SubmissionRecord : MarshalByRefObject, IEventSource
+        private class SubmissionRecord :
+#if FEATURE_MARSHALBYREFOBJECTS
```

Comment:
I'd just use FEATURE_APPDOMAIN for this (which is what I've been doing so far).  I don't think we need a separate feature flag for AppDomains and MarshalByRefObjects.